### ISSUE TITLE
Handling Jr. in generated cite keys

### ISSUE THREAD

**Issue by rolve** [ZotPlus/zotero-better-bibtex#236]
I generate my cite keys using the following pattern:

    [auth:fold][year][Title:nopunct:skipwords:select,1,1:fold:capitalize]

Now, I have some authors that contain ", Jr." in their last names (because there is no better place to put the Jr. in Zotero). This results in keys like `HalsteadJr.1985Multilisp` or `BakerJr.1977Incremental`, which is suboptimal.
I have tried the following pattern to strip the Jr. away:

    [auth:select,1,1:fold][year][Title:nopunct:skipwords:select,1,1:fold:capitalize]

However, this does not play well with last names like "von Praun", as it results in keys like `von2007Implicit`.
One way to solve this issue would be to always remove Jr. from author names, but I'm not sure this is a good idea. Another solution would be to let the user customize the `skipwords` list (either using some syntax or by proving an about:config option), so that one could use:

    [auth:skipwords:fold][year][Title:nopunct:skipwords:select,1,1:fold:capitalize]

**Comment by noschinl** [comment]
Customizing skipwords would also be nice for articles written in other languages than english.

**Comment by retorquere** [comment]
I've moved the skipwords to a preference, so you  should just be able to add `,jr` to the end. Note that skipwords is also applied to `title` and `shorttitle`.

**Comment by retorquere** [comment]
There isn't a UI for this btw, you can edit the string in `about:config`; the setting is `extensions.zotero.translators.better-bibtex.skipWords`.

**Comment by retorquere** [comment]
This wil be released as part of .12

**Comment by rolve** [comment]
Thank you for this! For anyone with the same problem: I had to use `:nopunct` to make it work:

    [auth:nopunct:skipwords:fold][year][Title:nopunct:skipwords:select,1,1:fold:capitalize]

**Comment by retorquere** [comment]
My pleasure -- I've added your example to the docs.

**Comment by rolve** [comment]
Somehow this has stopped working for me. I haven't used Better Bibtex in a long time, and now that I need it again, I realized that `Jr` is not stripped away anymore. In fact, a quick test showed that the `extensions.zotero.translators.better-bibtex.skipWords` setting seems to be ignored completely (I removed `the` from the list but it didn't show up in titles).

Unless I did something stupid (which is quite possible), I think this issue should be reopened.
